src/v/cluster/node/local_monitor.cc

```diff
@@ -135,31 +136,39 @@ void local_monitor::maybe_log_space_error(const storage::disk& disk) {
                                                    : alert_text);
 }
 
-void local_monitor::update_alert_state() {
-    _state.storage_space_alert = storage::disk_space_alert::ok;
-    for (const auto& d : _state.disks) {
+storage::disk_space_alert
+local_monitor::eval_disks(const std::vector<storage::disk>& disks) {
+    auto& cfg = config::shard_local_cfg();
+    unsigned alert_percent
+      = cfg.storage_space_alert_free_threshold_percent.value();
+    size_t alert_bytes = cfg.storage_space_alert_free_threshold_bytes.value();
+    size_t min_bytes = cfg.storage_min_free_bytes();
+
+    storage::disk_space_alert node_sa{};
+    for (const auto& d : disks) {
+        storage::disk_space_alert disk_sa{};
         vassert(d.total != 0.0, "Total disk space cannot be zero.");
-        // 1. Check alert thresholds.
-        auto [min_by_bytes, min_by_percent] = minimum_free_by_bytes_and_percent(
-          d.total);
-        auto alert_min = std::max(min_by_percent, min_by_bytes);
-        auto min_bytes = _min_free_bytes();
+        size_t min_by_percent = alert_percent_in_bytes(alert_percent, d.total);
+        auto alert_min = std::max(min_by_percent, alert_bytes);
         if (unlikely(d.free <= alert_min)) {
-            _state.storage_space_alert = storage::disk_space_alert::low_space;
+            disk_sa = storage::disk_space_alert::low_space;
         }
         // 2. Check degraded (read-only) threshold
         if (unlikely(d.free <= min_bytes)) {
-            _state.storage_space_alert = storage::disk_space_alert::degraded;
+            disk_sa = storage::disk_space_alert::degraded;
         }
-        maybe_log_space_error(d);
-        clusterlog.debug(
-          "alert % {}, alert bytes {}, min bytes {} disk.free {} -> alert {}",
-          min_by_percent,
-          min_by_bytes,
-          min_bytes,
-          d.free,
-          _state.storage_space_alert);
+        node_sa = std::max(node_sa, disk_sa);
     }
+    return node_sa;
+}
+
+// Preconditions: _state.disks is non-empty.
+void local_monitor::update_alert_state() {
+    auto space_alert = eval_disks(_state.disks);
+
+    _state.storage_space_alert = space_alert;
+    // TODO multi-disk support
+    maybe_log_space_error(_state.disks[0], space_alert);
```

Comment:
Could also just pass `disk.total` instead of `disk`.  You are right this is kinda awkward with some of these functions changed to static. Temporary until we do another health rpc format change (static so I can call from health monitor to recompute the non-serialized enum).  I had the same thought about a loop, but we don't support multiple disks yet.. so I was trying not to bloat the code unnecessarily. Its a bit awkward when we want to design APIs that support multiple disks but not waste cycles on something we know is a singleton. One of those things that depends on how obsessed with microoptimizations you are. I'm cool with a loop if folks don't mind the couple extra instructions/branches.

Reply:
Maybe if we don't support multiple disks, just omit it for clarity and brevity? Then when we plan to, all of that logic goes in a single commit

Reply:
There's a couple of places in the code where we do the "general design, specific implementation" thing of APIs that can support multiple disks, with implementations that don't. I was following the existing example ([link from older version of code](https://github.com/redpanda-data/redpanda/pull/3066/files#diff-83f77561273fcbd8c758ec86ddd985645aa2fcfd7a0fc687565f7bf92c585bacR59)) in metrics_reporter before I started refactoring and adding the support for disk space tracking.